### PR TITLE
WIP search: add an elasticsearch configuration tool

### PR DESCRIPTION
**Pull request by Letreguilly**
* NEW Adds an inveniomange command generating elasticsearch mapping
  from json schemas.

Signed-off-by: Sylvain Letreguilly <letreguilly@ill.fr>

### CODE REVIEW

File: invenio_search/config.py [hunk 1]
```diff
@@ -185,3 +185,12 @@
...
     "542__l": ["information_relating_to_copyright_status.copyright_status"],
 }
+
+#elasticsearch configuration tool parameters
+ES_ALL = 'true'
+ES_DATE_DETECTION = 'false'
+ES_NUMERIC_DETECTION = 'false'
+ES_DEFAULT_TYPE = "string"
+ES_DEFAULT_ANALYSER = "basic_analyser"
+ES_SCHEMA_FOLDER = "/home/letreguilly/virtualenvs/invenio/src/invenio-search/invenio_search/schemas/"
```
Comment (tiborsimko) [inline invenio_search/config.py]: WIP hard-coded paths will have to be amended

File: invenio_search/mapping.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+def create_es_mapping():
+    """main function, link jinja template and create the mapping"""
+    #settings
+    jsonInputSchemaFolder = cfg['ES_SCHEMA_FOLDER']
+    elasticMappingFile = cfg['ES_MAPPING_FILE']
+
+    generatedFieldFile = "/home/letreguilly/virtualenvs/invenio/src/invenio-search/invenio_search/generated/field.json"
+    AnalyserFile = "/home/letreguilly/virtualenvs/invenio/src/invenio-search/invenio_search/base/analyser.json"
```
Comment (nharraud) [inline invenio_search/mapping.py]: Absolute paths to change

File: invenio_search/mapping.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    env = Environment(loader=PackageLoader('invenio_search', '.'))
+
+    
+
+    #generate fields
+    rawFields = loadFieldsFromJson(jsonInputSchemaFolder)
+    formatedfields = format_fields(rawFields)
+    dumpsFields(formatedfields)
+    
+    jsonFields = dumpsFields(formatedfields)
```
Comment (nharraud) [inline invenio_search/mapping.py]: duplicated code (see line above)

**Comment by nharraud** [comment]
@jirikuncar I will continue this pull request later.

**Comment by jirikuncar** [comment]
Elasticsearch mapping generator will be released as separate package.

**Pull request closed without merging**